thread-manager/Cargo.toml

```diff
@@ -0,0 +1,38 @@
+[package]
+name = "agave-thread-manager"
+description = "Thread pool manager for agave"
+
+version = { workspace = true }
+authors = { workspace = true }
+repository = { workspace = true }
+homepage = { workspace = true }
+license = { workspace = true }
+edition = { workspace = true }
+
+publish = false
+
+[dependencies]
+anyhow = { workspace = true }
+cfg-if = { workspace = true }
+log = { workspace = true }
+num_cpus = { workspace = true }
+rayon = { workspace = true }
+serde = { workspace = true, features = ["derive"] }
+solana-metrics = { workspace = true }
+thread-priority = { workspace = true }
+tokio = { workspace = true, features = ["time", "rt-multi-thread"] }
+
+[target.'cfg(target_os = "linux")'.dependencies]
+affinity = "0.1.2"
```

Comment:
does this crate only exist for linux (is that a thing?), or just that we only use it for linux?

Reply:
It can compile for windows but I have no way to verify it works at all there. On mac it fails to build. I think it is better to only build it on linux.

Reply:
Sounds good! Just hadn't seen OS conditional dependencies before in the project